runtime/lua/vim/lsp/rpc.lua

```diff
@@ -608,6 +608,73 @@ function M.connect(host, port)
   end
 end
 
+---@class Dispatchers
+---@field notification function
+---@field server_request function
+---@field on_exit function
+---@field on_error function
+
+-- TODO: better name
+---@class StartServer
+---@field request fun(method: string, params: table?, callbackfun: fun(err: lsp.ResponseError | nil, result: any), notify_reply_callback:function?)
+---@field notify fun(method: string, params: any)
+---@field is_closing function
+---@field terminate function
+
+--- Create a LSP RPC client factory that connects via named pipes (window)
+--- or unix domain sockets (unix) to the given pipe_path (file path on
+--- Unix and name on Windows)
+---
+---@param pipe_path  string
+---@return fun(dispatchers: Dispatchers): StartServer?
```

Comment:
Feels a bit odd to have a "plural" type like `Dispatchers`. I find more intuitive to have a singular `vim.rpc.Dispatcher` type and make this parameter a table of those.

Reply:
I took theese types from the [private dispatchers](https://github.com/neovim/neovim/blob/master/runtime/lua/vim/lsp.lua#L1139-L1259) that are passed to this function by `vim.lsp.start_client`. Not all of them share the same signature, so this type can't be an array of `vim.lsp.rpc.Distpacher`, what do you suggest for this case?